tests/mir-opt/sroa.rs

```diff
@@ -70,7 +70,7 @@ fn f(a: *const u32) {
 
 pub fn escaping() {
     // Verify this struct is not flattened.
-    f(&Escaping { a: 1, b: 2, c: g() }.a);
+    f(&Escaping { a: 1, b: 2, c: g() } as *const Escaping as *const u32);
```

Comment:
```suggestion
    let e = Escaping { a: 1, b: 2, c: g() };
    f(&e.a);
```

> `f` uses the `&e.a` to access `e.c`. This is UB according to Miri today; however, T-opsem has not finalized that decision and as such rustc should not rely on it today. This test ensures that the `e` struct is not deaggregated into its fields, which would cause this to miscompile.